File: java/data/src/main/java/com/youknow/data/source/MoviesDataSource.java
package com.youknow.data.source;

import com.youknow.domain.repository.MoviesRepository;

public interface MoviesDataSource {

    void getNowPlaying(MoviesRepository.MoviesLoadedCallback callback);

    void getUpcoming(MoviesRepository.MoviesLoadedCallback callback);

    void getMovie(String movieId, MoviesRepository.MovieLoadedCallback callback);

}

File: kotlin-rx/data/src/main/java/com/youknow/data/source/MoviesDataSource.kt
package com.youknow.data.source

import com.youknow.domain.model.Movie
import com.youknow.domain.model.SimpleMovie
import io.reactivex.Single

interface MoviesDataSource {

    fun getNowPlaying(): Single<List<SimpleMovie>>

    fun getUpcoming(): Single<List<SimpleMovie>>

    fun getMovie(id: String): Single<Movie>

}